### PR TITLE
feat: adds handling for missing content with unit tests

### LINKED ISSUE #72: Add complete support for returning nullable response types
In the existing solution when the response from the API Call is missing and the return type is marked as nullable then the response handler does not return the `null` value but instead fails by throwing a deserialization exception.

Add the support for this above mentioned edge case, along with proper unit tests.

### PR DESCRIPTION
**Pull request by asadali214**
## What
This PR adds complete support for returning nullable response types using the nullability check of the generic `ResponseType` to set the response types as nullable when content is missing.

## Why
Previously response handler was throwing an exception when a response with no content is received.

Closes #72 

## Type of change
Select multiple if applicable.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause a breaking change)
- [X] Tests (adds or updates tests)
- [ ] Documentation (adds or updates documentation)
- [ ] Refactor (style improvements, performance improvements, code refactoring)
- [ ] Revert (reverts a commit)
- [ ] CI/Build (adds or updates a script, change in external dependencies)

## Dependency Change
N/A

## Breaking change
N/A

## Testing
Unit tests were added to test this feature

## Checklist
- [X] My code follows the coding conventions
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added new unit tests

### CODE REVIEW

File: APIMatic.Core/Response/ResponseHandler.cs [hunk 1]
```diff
@@ -138,6 +138,16 @@ internal ReturnType Result<ReturnType>(CoreContext<CoreRequest, CoreResponse> co
...
         }
 
+        private bool HasEmptyResponse(CoreResponse response)
+        {
+            var resType = typeof(ResponseType);
+            if (resType == typeof(VoidType))
+            {
+                return true;
+            }
+            return string.Equals(response.Body, string.Empty) && CoreHelper.IsNullableType(resType);
```
Comment (sufyankhanrao) [inline APIMatic.Core/Response/ResponseHandler.cs:148]: Blocker: As we agreed upon checking the whitespaces in the body as well therefore the response.Body should be trimmed before checking the equality here.

**Review by sufyankhanrao — DISMISSED** [review]

**Review by sufyankhanrao — APPROVED** [review]
